client/executor/runtime-test/src/lib.rs

```diff
@@ -337,3 +343,38 @@ sp_core::wasm_export_functions! {
 		return 1234;
 	}
 }
+
+// Returns a huge len. It should result in an error, and not an allocation.
+#[no_mangle]
+#[cfg(not(feature = "std"))]
+pub extern "C" fn test_return_huge_len(_params: *const u8, _len: usize) -> u64 {
+	pack_ptr_and_len(0, u32::MAX)
+}
+
+// Returns an offset right at the edge of the wasm memory boundary. With length 0, it should
+// succeed.
+#[no_mangle]
+#[cfg(not(feature = "std"))]
+pub extern "C" fn test_return_max_memory_offset(_params: *const u8, _len: usize) -> u64 {
+	// This should use `core::arch::wasm` instead of `core::arch::wasm32`,
+	// but `core::arch::wasm` depends on `#![feature(simd_wasm64)]` on current nightly.
+	// See https://github.com/Craig-Macomber/lol_alloc/issues/1
```

Comment:
I thought I added the first reference to "lol" in the substrate source, but running `rg`, it looks like there already was one. 🙃

Reply:
Yeah by me :P 

I think using `wasm32` is fine anyway, everything currently is build around wasm32. I think we can also remove the comment, not that useful :see_no_evil:

Reply:
Done. -2 `lol`s.

Reply:
:+1: